src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -646,6 +777,49 @@ void executeQueryWithParallelReplicasCustomKey(
         context, query_info.query, storage_id.getDatabaseName(), storage_id.getTableName(), /*table_function_ptr=*/nullptr);
     executeQueryWithParallelReplicasCustomKey(query_plan, storage_id, query_info, columns, snapshot, processed_stage, header, context);
 }
+
+bool canUseParallelReplicasOnInitiator(const ContextPtr & context)
```

Comment:
Could I kindly inquire about a matter? Why make a new function here to check if a cluster (shard) has only one node, rather than within the Context::canUseParallelReplicasOnInitiator()? If a cluster (shard) has only one node, all ParallelReplicas scenarios need to fall back to local execution, not just in buildQueryPlanForTableExpression(). @devcrafter Looking forward to your reply.

Reply:
Hi @zoomxi, it makes sense. If you're up to it, please feel free to provide a PR. Probably, a couple of requirements for such a case, -  there should be (1) a warning in traces about fallback to local execution, and (2)  a test to check this behavior.

Reply:
ok, thanks for your reply. @devcrafter